src/jit/gentree.cpp

```diff
@@ -12449,40 +12450,53 @@ GenTreePtr Compiler::gtFoldExprSpecial(GenTreePtr tree)
 //    the copy.
 //
 // Arguments:
-//    op  -- the box node to optimize
+//    op  - the box node to optimize
+//    options - controls whether and how trees are modified
+//        (see notes)
 //
 // Return Value:
-//    True if the upstream effects were removed. Note parts of the
-//    copy tree may remain, if the copy source had side effects.
-//
-//    False if the upstream effects could not be removed.
+//    A tree representing the original value to box, if removal
+//    is successful (but see note). nullptr if removal fails.
```

Comment:
FWIW, after reading this and the note I was still not sure what the return value is when `BR_DONT_REMOVE` gets passed... something like `successful/possible` instead of just `successful` here, or an explicit `Still returns the source tree if removal is legal` in the note for `BR_DONT_REMOVE`, would maybe have helped.

Reply:
Will do the former.